### PR TITLE
Added WebPush client documentation

### PR DESCRIPTION
**Pull request by ziccardi**
## Motivation
https://issues.jboss.org/browse/AEROGEAR-10134

## What
Added WebPush client configuration

### CODE REVIEW

File: website/docs/configuring_clients.md [hunk 1]
```diff
@@ -344,6 +344,238 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
+
+To be able to handle _push notifications_, a service worker is needed. In this guide we will provide a very simple implementation,
+for instructions about writing service workers please refer to the [official documentation](https://developer.mozilla.org/en-US/docs/Web/API/Push_API). 
+
+Inside the `public` folder, create a file named `simple-sw.js` with the following content:
+```javascript
+self.addEventListener('push', event => {
+  const data = event.data.text();
+  const msg_chan = new MessageChannel();
+  clients.matchAll().then(clients => {
```
Comment (secondsun) [inline website/docs/configuring_clients.md]: The first clients should probably be `self.clients` and the clients variable in the closure should be renamed so it doesn't conflict the with name of self.clients.  (If my eyeballing service worker code is right).

File: website/docs/configuring_clients.md [hunk 2]
```diff
@@ -344,6 +344,238 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
+  });
+});
+
+self.addEventListener('install', function(event) {
+  self.skipWaiting();
+});
+```
+
+:::important
+By default the react template unregisters the _service worker_ automatically. That must be removed.
```
Comment (secondsun) [inline website/docs/configuring_clients.md]: that must be disabled may sound more clear.

File: website/docs/configuring_clients.md [hunk 3]
```diff
@@ -344,6 +344,238 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
+:::
+
+Edit the `src/index.tsx` file and comment the `serviceWorker.unregister();` call.
+
+### Add _UnifiedPush Server Integration_
+
+#### Registration/Unregistration
+
+The object responsible for the registration/unregistration is the `PushRegistration` class, which exposes a `register` method.
+The `PushRegistration` constructors takes a configuration object as parameters with the following structure:
```
Comment (secondsun) [inline website/docs/configuring_clients.md]: constructor takes

File: website/docs/configuring_clients.md [hunk 4]
```diff
@@ -344,6 +344,238 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
+  url: 'your unified pushserver URL',
+  webpush: {
+    variantID: 'the ID of the WebPush variant you created in UPS',
+    variantSecret: 'the secret of the variant identified by variantID',
+    appServerKey: 'your VAPID public key (you can get it by opening the variant in UPS)'
+  }
+};
+
+/**
+ * Resister to the UPS
```
Comment (secondsun) [inline website/docs/configuring_clients.md]: Re*g*ister with

File: website/docs/configuring_clients.md [hunk 5]
```diff
@@ -344,6 +344,238 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
+  new PushRegistration(this.push_config)
+    .unregister()
+    .then(() => console.log("Unregistered successfully"))
+    .catch(error => console.log("Unregistration failed with error: ", error.message));
+};
+```
+
+#### Handling notifications
+
+To be able to receive notifications into your web application, you will need to register a callback that will receive
```
Comment (secondsun) [inline website/docs/configuring_clients.md]: in your web

File: website/docs/configuring_clients.md [hunk 6]
```diff
@@ -344,6 +344,238 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
+  const obj = JSON.parse(notification);
+  console.log('Notification: ', {
+    priority: obj.priority,
+    text: obj.alert,
+    date: new Date(),
+  });
+});
+```
+
+#### Putting all together
```
Comment (secondsun) [inline website/docs/configuring_clients.md]: Putting it all together

File: website/docs/configuring_clients.md [hunk 7]
```diff
@@ -344,6 +344,238 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
+      priority: obj.priority,
+      text: obj.alert,
+      date: new Date(),
+    });
+   });
+   this.register();
+}
+```
+
+The content of the `App.tsx` could be:
```
Comment (secondsun) [inline website/docs/configuring_clients.md]: /s/could/should

**Comment by ziccardi** [comment]
@secondsun Mind giving another look?

**Pull request merged**